feat: Draft goals (#7)
argoproj/argo-conformance-program: README.md

Status: Currently in draft and non-binding. 

# Argo Conformance Program
The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requriements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users need to know when they can they can trust these tools to work together and in a predictable way.

When users use an Argo certified offering they should be able to trust that offering to work with the open source versions of Argo and that configuration designed for Argo will always work.

Software conformance ensures predictability and helps protect the Argo brand. 

# Goals

> * Ensure a consistent and reliable user experience
> * Protect the Argo brand

Author: ```suggestion
* Ensure a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
```

From Henrik

> * Enable contributions from commercial partners
> * To grow and strengthen the Argo community

Author: ```suggestion
* Ensure contributions from commercial partners
* To grow and strengthen the Argo community
```

We will likely require active maintainers in order to meet conformance. It's not enough to encourage contributions, they should be required.

# Benefits of Certification
In addition to giving users confidence and protecting the Argo community certified offerings may

> * Advertise compliance and certification

Author: ```suggestion
* Advertise compliance and certification
* Use Argo logo and branding in marketing materials
```

Make sure that brand usage is explicitly stated.

> * Preference in community and marketing events etc.

Author: ```suggestion
* Exposure to community
```

Simplify language and avoid "preference".

Status: Currently in draft and non-binding. 

# Argo Conformance Program

> The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requriements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users need to know when they can they can trust these tools to work together and in a predictable way.

Reviewer: ```suggestion
+The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requirements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users can trust that products certified as compliant will interoperate with each other and be up-to-date with the latest open source releases. This ensures a consistent user experience between products and flexibility to choose between vendors.
```

Author: LGTM 👍

When users use an Argo certified offering they should be able to trust that offering to work with the open source versions of Argo and that configuration designed for Argo will always work.

Software conformance ensures predictability and helps protect the Argo brand. 

# Goals

> * Ensure a consistent and reliable user experience and API compatibility 
> * Protect the Argo brand
> * Ensure contributions from commercial partners
> * To grow and strengthen the Argo community

Author: ```suggestion
* Guarantee a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
* Ensure contributions from commercial partners
* Grow and strengthen the Argo community
```
Make lines consistent in form and don't reuse "ensure" too much.

> * Guarantee a consistent and reliable user experience and API compatibility 
> * Protect the Argo brand
> * Ensure contributions from commercial partners
> * Grow and strengthen the Argo community

Reviewer: ```suggestion
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 
+* Ensure contributions from commercial partners
 * Grow and strengthen the Argo community
+* Protect the Argo brand

```

Author: Good additions and the wording is solid. 👍

Status: Currently in draft and non-binding. 

# Argo Conformance Program
The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requirements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users can trust that products certified as compliant will interoperate with each other and be up-to-date with the latest open source releases. This ensures a consistent user experience between products and flexibility to choose between vendors.

When users use an Argo certified offering they should be able to trust that offering to work with the open source versions of Argo and that configuration designed for Argo will always work.

Software conformance ensures predictability and helps protect the Argo brand. 

# Goals
* Guarantee a consistent and predictable user experience
* Guarantee interoperability, including API compatibility
* Guarantee conformant products are up-to-date with latest Argo releases 

> * Promote contributions from commercial partners

Reviewer 2: What is the reason to change to Promote?

Reviewer: Is Encourage better?

Author: @edlee2121 for any additional changes please use the code suggest feature in a code review. Which will allow people to discuss the change before merging. 

We actually started with "Encourage" and the group felt that it wasn't strong enough as a goal and that we should basically require contributions to be coming back in some way. It's likely this will take the form of certified offerings to have a maintainer in the Argoproj. 

Given that I think we should go back to "Ensure".

Author: ```suggestion
* Ensure contributions from commercial partners
```

Reviewer 2: I like "Ensure" as the goal to really benefit the community in a measurable way. Thanks.

Reviewer: Sure not sure how to use that feature. Let me revert my commit and resubmit it.

When users use an Argo certified offering they should be able to trust that offering to work with the open source versions of Argo and that configuration designed for Argo will always work.

Software conformance ensures predictability and helps protect the Argo brand. 

# Goals
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 
+* Ensure contributions from commercial partners
 * Grow and strengthen the Argo community
+* Protect the Argo brand

# Benefits of Certification
In addition to giving users confidence and protecting the Argo community certified offerings may
* Advertise compliance and certification
* Use Argo logo and branding in marketing materials

> * Exposure to community

Reviewer: ```suggestion
* Exposure to community

# Benefits to Community
* Promote interoperability and compatibility between Argo products
* Discourage forking of the Argo code base
* Encourage commercial interests to work together to grow the Argo ecosystem rather than purely on competition with each other
```

Author: I like it!

Status: Currently in draft and non-binding. 

# Argo Conformance Program
+The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requirements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users can trust that products certified as compliant will interoperate with each other and be up-to-date with the latest open source releases. This ensures a consistent user experience between products and flexibility to choose between vendors.

When users use an Argo certified offering they should be able to trust that offering to work with the open source versions of Argo and that configuration designed for Argo will always work.

Software conformance ensures predictability and helps protect the Argo brand. 

# Goals
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 

> +* Ensure contributions from commercial partners

Reviewer 3: Do we want to use something softer, like "engagement" over contributions? As a goal, engagement feels better, then we can specify in the requirements exactly what it means

Author: I'm good with that @hblixt. @edlee2121 made a good point when I talked to him today about a race for people to get maintainers in. I think we can work it out in the requirements but I'm fine with either word.

Reviewer 3: I don't really feel strongly either way. It just felt a little more appropriate for a goal to use engagement over contributions, which seemed a bit more of a requirement.

Author: Debated this in committee with @alexec @wanghong230 and @hblixt decided to move forward with current language. May revisit if someone wants conformance without maintainers.

 * Grow and strengthen the Argo community
+* Protect the Argo brand

# Benefits of Certification
In addition to giving users confidence and protecting the Argo community certified offerings may
* Advertise compliance and certification
* Use Argo logo and branding in marketing materials
* Exposure to community

# Benefits to Community
* Promote interoperability and compatibility between Argo products
* Discourage forking of the Argo code base

> * Encourage commercial interests to work together to grow the Argo ecosystem rather than purely on competition with each other

Reviewer 3: We should leave out the negative part of the statement (ie "... rather than purely on competition with each other"). 
Focus on the positive :)

Author: ```suggestion
* Encourage commercial interests to work together to grow the Argo ecosystem
```

Author: Discussed in committee with @hblixt @alexec and @wanghong230 agreed to remove the last part as it's covered in the first part.
